src/Symfony/Component/Config/Resource/ReflectionClassResource.php

```diff
@@ -127,14 +127,35 @@ private function generateSignature(\ReflectionClass $class): iterable
         yield (int) $class->isFinal();
         yield (int) $class->isAbstract();
 
+        $enum = null;
+        if ($class->isEnum()) {
+            $enum = new \ReflectionEnum($class->name);
+        }
+
         if ($class->isTrait()) {
             yield print_r(class_uses($class->name), true);
+        } elseif ($enum) {
+            yield print_r(class_implements($class->name), true);
```

Comment:
can't we remove this "elseif" and rely on the "else" ? then maybe simplify a bit the added loop below since getConstants will return part of it already?